### ISSUE TITLE
Incompatibility with Python 3.10

### ISSUE THREAD

**Issue by EveCharbie** [pupil-labs/pupil#2222]
It should be written at the beginning of the procedures that you need a specific version of python.
It is sad when we get to the point where we have all set up our environment to find out on discord that we need to start over again with an environment in python 3.6 (on linux at least).
But still, you guys are rocking it ;)

**Comment by papr** [comment]
Hi @EveCharbie 

You should be able to run Pupil from source using newer Python versions than 3.6 on Linux, too. What is the exact issue that you ran into while setting up the environment and was solved by installing Python 3.6?

**Comment by EveCharbie** [comment]
I did not look into it carefully, but here is the error I got with python 3.10, which disappeared with python 3.6:

```
/home/user/Documents/Programmation/pupil/pupil_src/main.py:251: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  ipc_backbone_thread.setDaemon(True)
/home/user/Documents/Programmation/pupil/pupil_src/main.py:255: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  pull_pub.setDaemon(True)
/home/user/Documents/Programmation/pupil/pupil_src/main.py:259: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  log_thread.setDaemon(True)
/home/user/Documents/Programmation/pupil/pupil_src/main.py:263: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  delay_thread.setDaemon(True)
player - [ERROR] launchables.player: Process player_drop crashed with trace:
Traceback (most recent call last):
  File "/home/user/Documents/Programmation/pupil/pupil_src/launchables/player.py", line 854, in player_drop
    from pupil_recording import (
  File "/home/user/Documents/Programmation/pupil/pupil_src/shared_modules/pupil_recording/__init__.py", line 12, in <module>
    from .recording import PupilRecording
  File "/home/user/Documents/Programmation/pupil/pupil_src/shared_modules/pupil_recording/recording.py", line 29, in <module>
    class PupilRecording:
  File "/home/user/Documents/Programmation/pupil/pupil_src/shared_modules/pupil_recording/recording.py", line 67, in PupilRecording
    class FileFilter(collections.Sequence):
AttributeError: module 'collections' has no attribute 'Sequence'
```
If you feel it is a problem on my end, do not hesitate to close this issue since I already have a solution.

**Comment by papr** [comment]
Thank you for providing the traceback! Looks like the code is using deprecated functionality that was [removed in Python 3.10](https://docs.python.org/3/whatsnew/3.10.html#removed). In other words, the Pupil code base is currently not compatible with Python 3.10. I will update the documentation accordingly and look for a proper fix. Thanks!

**Comment by papr** [comment]
Fixed in #2251
